elastic/kibana
# [Semantic text] Make semantic text work with non-root level fields

## Summary

This makes semantic text work with non-root level reference fields. It also correctly adds copy_to to existing copy_to fields instead of replacing them, and streamlines a lot of the code.

To test these changes:

- Create an index
- Go to the index mappings page at `app/management/data/index_management/indices/index_details?{yourIndexName}=blah&tab=mappings`
- Add an object field with a text field inside
- Add a semantic text field referencing that text field
- If you're on a Macbook, create a new inference endpoint with the model `.elser_model_2` instead of using the default inference endpoint.
- Add a second semantic text field referencing that text field
- Save your mappings
- Use JSON view to verify that the newly created text field contains a `copy_to` field referencing both newly created semantic text fields
- Verify that the newly created semantic text fields are also in the JSON view



### Checklist

Delete any items that are not applicable to this PR.
